addon/actions/action.js

```diff
@@ -111,7 +115,7 @@ export default EmberObject.extend({
   requestUrl() {
     let modelName = this.get('modelName');
     let id = this.get('instance') ? this.get('model.id') : null;
-    let snapshot = this.get('model')._createSnapshot(this.get('config.adapterOptions'));
+    let snapshot = this.get('model')._internalModel.createSnapshot({ adapterOptions: this.get('config.adapterOptions') });
     let actionId = this.get('path');
```

Comment:
@Exelord
currently, path has to be explicitly passed as string. What do you think to get the function name (in model/resourceAction) as default (if path is not explicitly passed)?
arguments.callee.name

So e.g.:
publish: modelAction() -> it will result in path = publish

Reply:
you can not call callee on arguments. ;) Check out, warnings in console